master/internal/core.go

```diff
@@ -791,6 +793,95 @@ func (m *Master) restoreNonTerminalExperiments() error {
 	return nil
 }
 
+func (m *Master) restoreGenericTasks(ctx context.Context) error {
+	snapshots := []command.CommandSnapshot{}
+	err := db.Bun().NewSelect().Model(&snapshots).
+		Relation("Allocation").
+		Relation("Task").
+		Relation("Task.Job").
+		Where("allocation.end_time IS NULL").
+		Where("allocation.state != ?", model.AllocationStateTerminated).
+		Where("task.task_id = command_snapshot.task_id").
+		Where("command_snapshot.generic_task_spec IS NOT NULL").
+		Scan(ctx)
+	if err != nil {
+		return err
+	}
+
+	for i := range snapshots {
+		taskID := snapshots[i].TaskID
+		jobID := snapshots[i].Task.JobID
+
+		logCtx := logger.Context{
+			"job-id":    jobID,
+			"task-id":   taskID,
+			"task-type": snapshots[i].Task.TaskType,
+		}
+
+		priorityChange := func(priority int) error {
+			return nil
+		}
+		if err := tasklist.GroupPriorityChangeRegistry.Add(*jobID, priorityChange); err != nil {
+			return err
+		}
+
+		onAllocationExit := func(ae *task.AllocationExited) {
+			syslog := log.WithField("component", "genericTask").WithFields(logCtx.Fields())
```

Comment:
is there any nice way to DRY the parts of this that are shared with the normal, on launch exit func. trying to avoid someone adding something in one spot and missing it in the other.

Reply:
moved into helper function. Would it be worth cherry picking the commit into the main Generic task branch or just have the change done in this PR.

Reply:
depends on how you want to land it all. either is fine with me.